benchmarks/src/main/java/org/elasticsearch/benchmark/search/query/range/DateFieldMapperDocValuesSkipperBenchmark.java

```diff
@@ -0,0 +1,279 @@
+        final String hostName = "host-" + batchIndex;
+        final long timestampDelta = random.nextInt(0, deltaTime);
+        final long timestamp = BASE_TIMESTAMP + ((docIndex % batchSize) * deltaTime) + timestampDelta;
+
+        if (withDocValuesSkipper) {
+            doc.add(SortedNumericDocValuesField.indexedField(TIMESTAMP_FIELD, timestamp)); // NOTE: doc values skipper on `@timestamp`
+            doc.add(SortedDocValuesField.indexedField(HOSTNAME_FIELD, new BytesRef(hostName))); // NOTE: doc values skipper on `host.name`
+        } else {
+            doc.add(new LongPoint(TIMESTAMP_FIELD, timestamp)); // BKD tree on `@timestamp`
+            doc.add(new SortedNumericDocValuesField(TIMESTAMP_FIELD, timestamp)); // NOTE: doc values without the doc values skipper on
+                                                                                  // `@timestamp`
+            doc.add(new SortedDocValuesField(HOSTNAME_FIELD, new BytesRef(hostName))); // NOTE: doc values without the doc values skipper on
+                                                                                       // `host.name`
+        }
+
+        doc.add(new StringField(HOSTNAME_FIELD, hostName, Field.Store.NO));
+    }
+
+    /**
+     * Computes a dynamic timestamp upper bound based on the batch size,
+     * timestamp increment, and user-specified fraction.
+     *
+     * @return The computed upper bound for the timestamp range query.
+     */
+    private long rangeEndTimestamp() {
+        return BASE_TIMESTAMP + ((long) (batchSize * deltaTime * queryRange));
+    }
+
+    @Benchmark
+    public void rangeQueryWithoutDocValuesSkipper(final Blackhole bh) throws IOException {
+        bh.consume(rangeQuery(indexSearcherWithoutDocValuesSkipper, BASE_TIMESTAMP, rangeEndTimestamp(), true));
+    }
+
+    @Benchmark
+    public void rangeQueryWithDocValuesSkipper(final Blackhole bh) throws IOException {
+        bh.consume(rangeQuery(indexSearcherWithDocValuesSkipper, BASE_TIMESTAMP, rangeEndTimestamp(), false));
+    }
+
+    private long rangeQuery(final IndexSearcher searcher, long rangeStartTimestamp, long rangeEndTimestamp, boolean isIndexed)
+        throws IOException {
+        assert rangeEndTimestamp > rangeStartTimestamp;
+        final Query rangeQuery = isIndexed
+            ? new IndexOrDocValuesQuery(
+                LongPoint.newRangeQuery(TIMESTAMP_FIELD, rangeStartTimestamp, rangeEndTimestamp),
+                SortedNumericDocValuesField.newSlowRangeQuery(TIMESTAMP_FIELD, rangeStartTimestamp, rangeEndTimestamp)
+            )
+            : SortedNumericDocValuesField.newSlowRangeQuery(TIMESTAMP_FIELD, rangeStartTimestamp, rangeEndTimestamp);
+        final Query query = new IndexSortSortedNumericDocValuesRangeQuery(
+            TIMESTAMP_FIELD,
+            rangeStartTimestamp,
+            rangeEndTimestamp,
+            rangeQuery
+        );
+        return searcher.search(query, nDocs, QUERY_SORT).totalHits.value();
```

Comment:
I will use `count` here instead of `search` to try to rule-out the work we do when collecting documents. The expectation si that, no matter if we use doc values sparse indices or not, in the fetch phase we need to fetch the same set of documents which are laid-out on disk in the same way in both scenarios. Hopefully using `count` will allow us to isolate the search phase only in the flame graph.